TST: Windows test failed with Microsoft Visual C++ 14.0 or greater is required

Just started happening. Not sure why.

Example log: https://github.com/astropy/pytest-doctestplus/runs/5327364400?check_suite_focus=true

```
_____________ ERROR collecting build/lib.win-amd64-3.8/module2.py _____________
build\lib.win-amd64-3.8\module2.py:1: in <module>
    from _module2 import foo
E   ModuleNotFoundError: No module named '_module2'
!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
============================== 1 error in 0.21s ===============================
---------------------------- Captured stderr call -----------------------------
error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
```

Why are we building C-ext in the test suite here, @lpsinger ?

https://github.com/astropy/pytest-doctestplus/blob/feea9835dd345d2efc9907a4be9f92b0fbe37289/tests/test_doctestplus.py#L951

--- a/.github/workflows/python-tests.yml
+++ b/.github/workflows/python-tests.yml
@@ -20,7 +20,7 @@ jobs:
           - os: ubuntu-latest
             python-version: 3.7
             toxenv: py37-test-pytest46
-          - os: windows-latest
+          - os: windows-2019
             python-version: 3.7
             toxenv: py37-test-pytest50
           - os: macos-latest
@@ -29,7 +29,7 @@ jobs:
           - os: ubuntu-latest
             python-version: 3.7
             toxenv: py37-test-pytest52
-          - os: windows-latest
+          - os: windows-2019
             python-version: 3.8
             toxenv: py38-test-pytest53
           - os: ubuntu-latest

